unittests/comprehensive/test_iLQRHalfCheetah.cpp

```diff
@@ -41,195 +40,6 @@ using namespace realtime;
-  auto simulate_traj = [&](std::vector<Eigen::VectorXs> X, std::vector<Eigen::VectorXs> U, bool render)
-  {
-    world->setState(init_state);
-    if(render)
-    {
-      server.renderWorld(world);
-      for(int i = 0; i < steps-1; i++)
-      {
-        rollout.getPoses().col(i) = world->getPositions();
-        rollout.getVels().col(i) = world->getVelocities();
-        rollout.getControlForces().col(i) = world->mapToForceSpaceVector(U[i]);
-        world->setAction(U[i]);
-        world->step();
-        X[i+1] = world->getState();
-        server.renderWorld(world);
-        usleep(10000);
-      }
-      rollout.getPoses().col(steps-1) = world->getPositions();
-      rollout.getVels().col(steps-1) = world->getVelocities();
-      s_t loss = costFn->computeLoss(&rollout);
-      return loss;
-    }
-    else
-    {
-      for(int i = 0; i < steps-1; i++)
-      {
-        rollout.getPoses().col(i) = world->getPositions();
-        rollout.getVels().col(i) = world->getVelocities();
-        rollout.getControlForces().col(i) = world->mapToForceSpaceVector(U[i]);
-        world->setAction(U[i]);
-        world->step();
-        X[i+1] = world->getState();
-      }
-      rollout.getPoses().col(steps-1) = world->getPositions();
-      rollout.getVels().col(steps-1) = world->getVelocities();
-      s_t loss = costFn->computeLoss(&rollout);
-      return loss;
-    }
-  };
-  // Instead of starting a single thread, iLQR Trajectory optimization from starting state
-  s_t init_cost = simulate_traj(ilqr.getStatesFromiLQRBuffer(), ilqr.getActionsFromiLQRBuffer(), false);
-  std::cout << "Initial Cost: " << init_cost << std::endl;
-  ilqr.setCurrentCost(init_cost);
-  s_t prev_cost = 1e10;
-  s_t threshold = 0.01;
-  
-  // Print out current parameters settings
-  std::cout << "Alpha: " << ilqr.getAlpha() << "\n"
-            << "MU: " << ilqr.getMU() << std::endl;
-  
-  int iter = 0;
-  while(iter < maxIter)
-  {
-    // Set the world to initial state
-    world->setState(init_state);
-    
-    bool forwardFlag = ilqr.ilqrForward(world);
-    bool backwardFlag = false;
-    if(!forwardFlag)
-    {
-      std::cout << "Optimization Terminated, Exiting ..." <<std::endl;
-      break;
-    }
-    else
-    {
-      backwardFlag = ilqr.ilqrBackward();
-    }
-    std::cout << "Iteration: " << iter+1 << " Cost: " << ilqr.getCurrentCost() << std::endl;
-    if(!backwardFlag)
-    {
-      std::cout << "Backward Terminated, Exiting ..." << std::endl;
-      break; 
-    }
-    if(abs(prev_cost-ilqr.getCurrentCost()) < threshold)
-    {
-      std::cout << "Optimization Converged, Existing ..." << std::endl;
-      break;
-    }
-    prev_cost = ilqr.getCurrentCost();
-    iter++;
-  }
-
-  // Demonstrate the performance
-  s_t final_cost = 0;
-  for(int i = 0; i < 5; i++)
-  {
-    final_cost = simulate_traj(ilqr.getStatesFromiLQRBuffer(), ilqr.getActionsFromiLQRBuffer(), true);
-  }
-  std::cout << "Final Cost: " << final_cost << std::endl;
```

Comment:
We should put a `EXPECT_TRUE(final_cost < some_value);` here, where you replace `some_value` with a number that makes the test barely pass.